### PR TITLE
Add cache_from as an available command

### PR DESCRIPTION
**Pull request by jyapayne**
This fixes the issues with PR #4310.

Fixes: #4309

Signed-off-by: Joey Payne <jyapayne@gmail.com>

### CODE REVIEW

**Comment by jyapayne** [comment]
@shin- My jenkins build failed because the docker version in the Dockerfile is too low. In the Dockerfile, your repository lists

```dockerfile
RUN curl https://get.docker.com/builds/Linux/x86_64/docker-1.8.3 \
        -o /usr/local/bin/docker && \
    chmod +x /usr/local/bin/docker
```

as the method of installing docker. I realize that this is to keep the image lighter. Is there a way to update this docker version to the latest (1.13.1)?

**Comment by shin-** [comment]
The Dockerfile does not need to be changed - the CLI version is irrelevant. You do need to use API version 1.25, which means the `cache_from` option can only be added to format 3.0+

**Comment by jyapayne** [comment]
@shin- Ah, okay. I've updated the test and then schema to reflect the information you've provided. I'll close those other issues.

**Comment by shin-** [comment]
@jyapayne Do you mind rebasing this against the newest `master`? That should resolve the Jenkins issue hopefully.

File: compose/config/config_schema_v3.0.json [hunk 1]
```diff
@@ -60,7 +60,8 @@
               "properties": {
                 "context": {"type": "string"},
                 "dockerfile": {"type": "string"},
-                "args": {"$ref": "#/definitions/list_or_dict"}
+                "args": {"$ref": "#/definitions/list_or_dict"},
+                "cache_from": {"$ref": "#/definitions/list_of_strings"}
```
Comment (dnephin) [inline compose/config/config_schema_v3.0.json]: Config v3.0 is frozen, so it can't be added here.

File: compose/config/config.py [hunk 2]
```diff
@@ -101,6 +101,7 @@
 
 ALLOWED_KEYS = DOCKER_CONFIG_KEYS + [
     'build',
+    'cache_from',
```
Comment (dnephin) [inline compose/config/config.py]: `cache_from` is under `build`, so it shouldn't be in this list. This list is only for the top level service keys

File: compose/config/config_schema_v3.0.json (hunk 1 above)
Comment (jyapayne) [inline compose/config/config_schema_v3.0.json]: Alright, so just to v3.1 then. I'll remove this.

File: compose/config/config.py (hunk 2 above)
Comment (jyapayne) [inline compose/config/config.py]: Alright, I'll remove it!

**Comment by jyapayne** [comment]
@shin- rebased and issues mentioned fixed! Hopefully the build passes this time.

**Comment by shin-** [comment]
Thank you!

File: compose/config/config_schema_v3.1.json [hunk 3]
```diff
@@ -71,7 +71,8 @@
               "properties": {
                 "context": {"type": "string"},
                 "dockerfile": {"type": "string"},
-                "args": {"$ref": "#/definitions/list_or_dict"}
+                "args": {"$ref": "#/definitions/list_or_dict"},
+                "cache_from": {"type": "#/definitions/list_of_strings"}
```
Comment (dnephin) [inline compose/config/config_schema_v3.1.json]: This doesn't seem to work. I think this should be `$ref` ?

Comment (jyapayne) [inline compose/config/config_schema_v3.1.json]: Should I open a new PR then?

Comment (dnephin) [inline compose/config/config_schema_v3.1.json]: Nope, it should be fixed by #4616

**Comment by cerdiogenes** [comment]
I'm receiving `unsupported option: 'cache_from'`, is it supported in `docker-compose version 1.13.0, build 1719ceb`?

**Comment by shin-** [comment]
Only for the following Compose file formats: 2.2, 3.2 and 3.3

**Comment by cerdiogenes** [comment]
@shin- when using `docker-compose.override.yml` the cache_from options is ignored, is this a expected behavior?

**Comment by shin-** [comment]
We fixed that bug in the 1.14 RC.

**Pull request merged**